src/ghcup/script.rs

```diff
@@ -24,18 +22,12 @@ impl SnapshotStorage<SnapshotMeta> for GhcupScript {
     ) -> Result<Vec<SnapshotMeta>> {
         let logger = mission.logger;
         let progress = mission.progress;
-        let client = mission.client;
 
         info!(logger, "fetching metadata of ghcup install script...");
         progress.set_message("fetching head of url");
-        let last_modified = get_last_modified(&client, self.script_url.as_str()).await?;
 
         progress.finish_with_message("done");
-        Ok(vec![SnapshotMeta {
-            key: String::from("install.sh"),
-            last_modified,
-            ..Default::default()
-        }])
+        Ok(vec![SnapshotMeta::new(String::from("install.sh"))])
```

Comment:
This won't work. This `SnapshotMeta`, created with `SnapshotMeta::new(String::from("install.sh"))`, don't have `last_modified` inside. When diff this object with s3 object, last_modified won't be compared.